addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -109,6 +109,7 @@ def _compute_message_unread(self):
                       WHERE mail_message.model = 'discuss.channel'
                         AND mail_message.message_type NOT IN ('notification', 'user_notification')
                         AND mail_message.id >= discuss_channel_member.new_message_separator
+                        AND mail_message.body != ''
```

Comment:
That's not enough to determine body is semantically empty thus message is considered empty.

We have a `MailMessage._filter_empty()`, probably need to post-process query results with that function to pick elligibly "non-empty" messages.

Reply:
We can't post-process an optimized query like this one, it has to be done in SQL I'm afraid.
We can't load all the messages.

Reply:
(in master) We really should just either delete message forever or starting displaying "message was deleted" because the current situation of invisible messages is far too problematic to handle everywhere in code.

Reply:
To my knowledge message deletion on Discuss only happens by setting the message body to an empty string so in the case of Discuss chats I believe it should be enough but I might be wrong.